bundles/core/src/main/java/com/adobe/cq/commerce/demandware/preview/PreviewServiceImpl.java

```diff
@@ -217,77 +194,75 @@ private String getPreviewContent(final String endPoint, final Resource resource,
             HttpClientUtils.closeQuietly(responseObj);
             HttpClientUtils.closeQuietly(httpClient);
         }
-
+        
         if (StringUtils.isEmpty(renderedPreview)) {
             renderedPreview = ERROR_MSG;
         }
-
+        
         return renderedPreview;
     }
-
-    private String getInstanceEndPoint() {
-        return clientProvider.getDefaultClient().getEndpoint();
+    
+    private DemandwareClient getDemandwareClient(Page containingPage) {
+        // call Demandware to render preview for component
+        Optional<DemandwareClient> client = clientProvider.getClientForSpecificInstance(instanceId.getInstanceId(containingPage));
+        if(client.isPresent()){
+            return client.get();
+        }else{
+            LOG.error("FailedtogetDemandwareClientfor[{}]DemandwareinstanceId.",instanceId.getInstanceId(containingPage));
```

Comment:
if DemandwareClient will not be found the error will be logged in clientProvider.getClientForSpecificInstance - no need to duplicate logging